src/core/stdcpp/allocator.d

```diff
@@ -30,35 +25,114 @@ struct allocator(T)
     static assert(!is(T == const), "The C++ Standard forbids containers of const elements because allocator!(const T) is ill-formed.");
     static assert(!is(T == immutable), "immutable is not representable in C++");
     static assert(!is(T == class), "Instantiation with `class` is not supported; D can't mangle the base (non-pointer) type of a class. Use `extern (C++, class) struct T { ... }` instead.");
+extern(D):
+
+    ///
+    this(U)(ref allocator!U) {}
 
     ///
     alias value_type = T;
 
+    ///
+    alias rebind(U) = allocator!U;
+
     version (CppRuntime_Microsoft)
     {
-        ///
-        T* allocate(size_t count) @nogc;
-        ///
-        void deallocate(T* ptr, size_t count) @nogc;
+        import core.stdcpp.xutility : _MSC_VER;
 
         ///
-        enum size_t max_size = size_t.max / T.sizeof;
+        T* allocate(size_t count) @nogc
+        {
+            static if (_MSC_VER <= 1800)
+            {
+                import core.stdcpp.xutility : _Xbad_alloc;
+                if (count == 0)
+                    return null;
+                T* mem;
+                if ((size_t.max / T.sizeof < count) || (mem = __cpp_new(count * T.sizeof)) == 0)
```

Comment:
```
src\core\stdcpp\allocator.d(52): Error: cannot implicitly convert expression `__cpp_new(count * 4LU)` of type `void*` to `int*`
test\stdcpp\src\allocator_test.d(17): Error: template instance `core.stdcpp.allocator.allocator!int` error instantiating
src\core\stdcpp\allocator.d(52): Error: cannot implicitly convert expression `__cpp_new(count * 8LU)` of type `void*` to `double*`
test\stdcpp\src\allocator_test.d(18): Error: template instance `core.stdcpp.allocator.allocator!double` error instantiating
src\core\stdcpp\allocator.d(52): Error: cannot implicitly convert expression `__cpp_new(count * 24LU)` of type `void*` to `MyStruct*`
test\stdcpp\src\allocator_test.d(19): Error: template instance `core.stdcpp.allocator.allocator!(MyStruct)` error instantiating
```

```suggestion
-                 if ((size_t.max / T.sizeof < count) || (mem = __cpp_new(count * T.sizeof)) == 0)
+                if ((size_t.max / T.sizeof < count) || (mem = cast(T*)__cpp_new(count * T.sizeof)) is null)
```

Reply:
I can see that ;) ... i just had to wait 2 hours to get the error message!

Reply:
> I can see that ;) ... i just had to wait 2 hours to get the error message!

If you cannot test locally, maybe it would be more productive to setup your own CI on appveyor, you can select an image with VS2013 installed, too (see top of appveyor.xml). You can probably set it up before next iteration here is done...

Reply:
I pushed before going to work. if it fails again, i'll do that tonight.
I had no idea appveyor had testers for 2013 still. we should just use it for all windows builds and get rid of the win32 auto-tester machines.

Look at the auto-tester logs; most machines take 5-10 minutes, except the windows machines which take 30-50 miunutes...